routers/repo/editor.go

```diff
@@ -161,7 +161,8 @@ func NewFile(ctx *context.Context) {
 func editFilePost(ctx *context.Context, form auth.EditRepoFileForm, isNewFile bool) {
 	canCommit := renderCommitRights(ctx)
 	treeNames, treePaths := getParentTreeFields(form.TreePath)
-	branchName := ctx.Repo.BranchName
+	branchName := path.Clean("/" + ctx.Repo.BranchName)[1:]
```

Comment:
I think this validation should be used for this:
https://github.com/go-gitea/gitea/blob/d056bf300ff5ebd89d8b0035722c94a3b08ac745/modules/validation/binding.go#L39

Reply:
It appears that that is already  bound:

https://github.com/go-gitea/gitea/blob/76408d50fb338e9239ee06bb26eec28453167300/modules/auth/repo_form.go#L551-L560

Reply:
than it shouldn't be problem imho

Reply:
I've removed the path.Clean and added bindings to the other users of repofiles.